cmd/listen.go

```diff
@@ -64,7 +66,7 @@ func processListener(interfaces *[]string, lp []string, promisc bool, bpf_filter
 			ports:   ports,
 			ipaddr:  ipaddr,
 			timeout: to,
-			promisc: promisc,
+			promisc: netif.Flags&net.FlagBroadcast != 0, // has broadcast, so promisc = true
```

Comment:
nit: just assign the right side to a variable to make it easier to read.  Also, I can never remember operator priority so some parens would be nice :)